Source/Pawnmorphs/Esoteria/Thoughts/Worker_BeastMaster.cs

```diff
@@ -23,41 +24,16 @@ public class Worker_BeastMaster : ThoughtWorker
         /// <returns></returns>
         protected override ThoughtState CurrentStateInternal(Pawn p)
         {
-            _animalNames.Clear();
-            GetAnimals(p, _animalNames);
-            if (_animalNames.Any())
-                return false;
-            if (_animalNames.Count == 1)
-                return ThoughtState.ActiveAtStage(0, _animalNames[0]);
-            return ThoughtState.ActiveAtStage(1, _animalNames.ToCommaList(true));
-        }
-
-        private bool AnimalMasterCheck(Pawn p, Pawn animal)
-        {
-            return animal.playerSettings.RespectedMaster == p;
-        }
-
-        private void GetAnimals([NotNull] Pawn p, [NotNull] List<string> outAnimals)
-        {
-            outAnimals.Clear();
-            foreach (Pawn pawn in PawnsFinder.AllMapsWorldAndTemporary_Alive)
+            if (def.IsValidFor(p))
             {
-                if (pawn.playerSettings?.Master != p) continue;
-                if (!pawn.IsFormerHuman()) continue;
-
-                if (pawn.Faction != p.Faction) continue;
-
-                if (AnimalMasterCheck(p, pawn))
-                    outAnimals.Add(pawn.Name?.ToStringFull ?? pawn.LabelShort);
+                return base.CurrentStateInternal(p);
             }
+            return false;
         }
 
-
-        private int GetAnimalsCount(Pawn p)
+        protected override bool AnimalMasterCheck(Pawn p, Pawn animal)
         {
-            _animalNames.Clear();
-            GetAnimals(p, _animalNames);
-            return _animalNames.Count;
+            return animal.playerSettings.RespectedMaster == p && animal.IsFormerHuman();
```

Comment:
"animal.playerSettings.RespectedMaster == p" can actually be shifted to vanilla by replacing it with "base.AnimalMasterCheck(p, animal)" since that is the entirety of base function.

Reply:
best to use base.AnimalMasterCheck in case any other mods patch that method

Reply:
Adjusted